src/engine/enginemaster.cpp

```diff
@@ -706,19 +949,73 @@ void EngineMaster::onOutputDisconnected(AudioOutput output) {
             // not used, because we need the master buffer for headphone mix
             // and recording/broadcasting as well
             break;
+        case AudioOutput::BOOTH:
+            m_pBoothEnabled->forceSet(0.0);
+            break;
         case AudioOutput::HEADPHONES:
-            m_pHeadphoneEnabled->set(0.0);
+            m_pHeadphoneEnabled->forceSet(0.0);
             break;
         case AudioOutput::BUS:
             m_bBusOutputConnected[output.getIndex()] = false;
             break;
         case AudioOutput::DECK:
             // We don't track enabled decks.
             break;
-        case AudioOutput::SIDECHAIN:
+        case AudioOutput::RECORD_BROADCAST:
             // We don't track enabled sidechain.
             break;
         default:
             break;
     }
 }
+
+void EngineMaster::onInputConnected(AudioInput input) {
+    switch (input.getType()) {
+      case AudioInput::MICROPHONE:
+          m_pNumMicsConfigured->set(m_pNumMicsConfigured->get() + 1);
```

Comment:
SoundManager::setConfig calls SoundManager::closeDevices, which now calls EngineMaster::onInputDisconnected which decrements this CO for each mic disconnected, before SoundManager initializes a new configuration with SoundManager::setupDevices.